workerapi/client/client.go

```diff
@@ -143,14 +122,16 @@ func (c *simpleClient) StatusAll() ([]runner.RunStatus, runner.ServiceStatus, er
 	if ws.Error != "" {
 		svcErr = errors.New(ws.Error)
 	}
-	return ws.Runs, runner.ServiceStatus{Initialized: ws.Initialized, Error: svcErr}, nil
-}
-
-func (c *simpleClient) QueryNow(q runner.Query) ([]runner.RunStatus, runner.ServiceStatus, error) {
-	st, svc, err := c.StatusAll()
+	st := ws.Runs
+	svc := runner.ServiceStatus{Initialized: ws.Initialized, Error: svcErr}
 	if err != nil {
 		return nil, runner.ServiceStatus{}, err
 	}
 	st, err = runners.StatusesRO(st).QueryNow(q)
 	return st, svc, err
 }
+
+func (c *simpleClient) Query(q runner.Query, w runner.Wait) ([]runner.RunStatus, runner.ServiceStatus, error) {
+	// TODO: implement
```

Comment:
?

Reply:
Whoops - had to add this to collapse the StatusQueryNower iface into the StatusQuerier, even though we don't use this. Meant to add a better comment.